hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/json/JsonValueFunction.java

```diff
@@ -0,0 +1,203 @@
+        this.onError = onError;
+    }
+
+    public static JsonValueFunction<?> create(Expression<?> json,
+                                              Expression<?> path,
+                                              Expression<?> defaultValue,
+                                              QueryDataType resultType,
+                                              SqlJsonValueEmptyOrErrorBehavior onEmpty,
+                                              SqlJsonValueEmptyOrErrorBehavior onError) {
+        final Expression<?>[] operands = new Expression<?>[] {
+                json,
+                path,
+                defaultValue
+        };
+        return new JsonValueFunction<>(operands, resultType, onEmpty, onError);
+    }
+
+    @Override
+    public int getFactoryId() {
+        return JetSqlSerializerHook.F_ID;
+    }
+
+    @Override
+    public int getClassId() {
+        return JetSqlSerializerHook.JSON_VALUE;
+    }
+
+    @Override
+    public T eval(final Row row, final ExpressionEvalContext context) {
+        final Object operand0 = operands[0].eval(row, context);
+        final String json = operand0 instanceof HazelcastJsonValue
+                ? operand0.toString()
+                : (String) operand0;
+        final String path = (String) operands[1].eval(row, context);
+        final Object defaultValue = operands[2].eval(row, context);
+
+        if (isNullOrEmpty(path)) {
+            return onErrorResponse(onError, QueryException.error("JSON_VALUE path expression is empty"), defaultValue);
+        }
+
+        if (isNullOrEmpty(json)) {
+            return onEmptyResponse(onEmpty, defaultValue);
+        }
+
+        try {
+            return execute(json, path);
+        } catch (Exception exception) {
+            return onErrorResponse(onError, exception, defaultValue);
+        }
+    }
+
+    private T onEmptyResponse(SqlJsonValueEmptyOrErrorBehavior onEmpty, Object defaultValue) {
+        switch (onEmpty) {
+            case ERROR:
+                throw QueryException.error("JSON argument is empty.");
+            case DEFAULT:
+                return (T) defaultValue;
+            case NULL:
+            default:
+                return null;
+        }
+    }
+
+    private T onErrorResponse(SqlJsonValueEmptyOrErrorBehavior onError, Exception exception, Object defaultValue) {
+        switch (onError) {
+            case ERROR:
+                throw QueryException.error("JSON_VALUE failed: ", exception);
+            case DEFAULT:
+                return (T) defaultValue;
+            case NULL:
+            default:
+                return null;
+        }
+    }
+
+    private T execute(final String json, final String path) {
+        final Object result = JsonPathUtil.read(json, path);
+        if (JsonPathUtil.isArrayOrObject(result)) {
+            throw QueryException.error("Result of JSON_VALUE can not be array or object.");
+        }
+        return (T) convertResultType(result);
```

Comment:
We do multiple conversions here for numbers. First, `JsonPathUtil.read` converts to BigDecimal or BigInteger. Then, for integer, we convert to `long` to see the range and then to a smaller type, if it fits.

We should directly convert to the target type.